components/Habit.js

```diff
@@ -3,99 +3,81 @@ import EmojiModal from "./EmojiModal";
 import InlineEdit from "./InlineEdit";
 
 export default function Habit(props) {
-  const [editingValue, setEditingValue] = useState(props.habit.text);
-  const [checkmark, setCheckmark] = useState("/grayCheckMark.svg");
-  const [isEmojiModalShowing, setIsEmojiModalShowing] = useState(false);
+  const { text } = props.habit;
+  const {
+    id,
+    deleteHandler,
+    updateCompleteStatus,
+    updateHabitTextHandler,
+    updateHabitEmojiHandler,
+  } = props;
+  const [editingValue, setEditingValue] = useState(text);
+  const [isModalVisible, setIsModalVisible] = useState(false);
   const [emoji, setEmoji] = useState(props.habit.emoji);
-  const [textDecoration, setTextDecoration] = useState("none");
+  const textDecoration = props.habit.completed ? "line-through" : "none";
+  const checkmark = props.habit.completed
+    ? "/redCheckMark.svg"
+    : "/grayCheckMark.svg";
 
-  const deleteHandler = (e) => {
+  const onDeleteHabit = (e) => {
     e.stopPropagation();
-    props.deleteHandler(props._id);
+    deleteHandler(id);
   };
 
-  const completeHabit = () => {
-    if (props.habit.completed === false) {
-      props.editCompletedHabit(props._id, true);
-      setCheckmark("/redCheckMark.svg");
-      setTextDecoration("line-through");
-    } else {
-      props.editCompletedHabit(props._id, false);
-      setCheckmark("/grayCheckMark.svg");
-      setTextDecoration("none");
-    }
+  const onHabitCardClick = () => {
```

Comment:
I don't need to pass in the `habit.completed` prop. I can toggle update the completion status of the habit on my database with just the id. Everything works the same